homeassistant/components/sabnzbd/__init__.py

```diff
@@ -1,234 +1,58 @@
-        key="queue_size",
-        name="Queue",
-        native_unit_of_measurement=DATA_MEGABYTES,
-        field_name="mb",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="queue_remaining",
-        name="Left",
-        native_unit_of_measurement=DATA_MEGABYTES,
-        field_name="mbleft",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="disk_size",
-        name="Disk",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="diskspacetotal1",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="disk_free",
-        name="Disk Free",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="diskspace1",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="queue_count",
-        name="Queue Count",
-        field_name="noofslots_total",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="day_size",
-        name="Daily Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="day_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="week_size",
-        name="Weekly Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="week_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="month_size",
-        name="Monthly Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="month_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="total_size",
-        name="Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="total_size",
-    ),
+from .const import (
+    ATTR_SPEED,
+    DEFAULT_SPEED_LIMIT,
+    DOMAIN,
+    KEY_API,
+    KEY_NAME,
+    SERVICE_PAUSE,
+    SERVICE_RESUME,
+    SERVICE_SET_SPEED,
+    SIGNAL_SABNZBD_UPDATED,
+    UPDATE_INTERVAL,
 )
+from .sab import get_client
 
-SENSOR_KEYS: list[str] = [desc.key for desc in SENSOR_TYPES]
+PLATFORMS = ["sensor"]
+_LOGGER = logging.getLogger(__name__)
 
 SPEED_LIMIT_SCHEMA = vol.Schema(
     {vol.Optional(ATTR_SPEED, default=DEFAULT_SPEED_LIMIT): cv.string}
 )
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_API_KEY): cv.string,
-                vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-                vol.Optional(CONF_PATH): cv.string,
-                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-                vol.Optional(CONF_SENSORS): vol.All(
-                    cv.ensure_list, [vol.In(SENSOR_KEYS)]
-                ),
-                vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
-
 
-async def async_check_sabnzbd(sab_api):
-    """Check if we can reach SABnzbd."""
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the SabNzbd Component."""
+    sab_api = await get_client(hass, entry.data)
```

Comment:
This should raise `ConfigEntryNotReady` errors if it can't connect to the host, and if authentication fails, it should log an error and `return False`.

Reply:
done